src/lxd.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""LXD client interface.
+
+The Lxd class is intend to be layer of abstraction to isolate the underlying implementation of LXD.
+"""
+from __future__ import annotations
+
+import io
+import tempfile
+from subprocess import CalledProcessError  # nosec B404
+from typing import IO, Optional, Tuple, Union
+
+import pylxd.models
+
+from errors import LxdError
+from lxd_type import LxdInstanceConfig, ResourceProfileConfig, ResourceProfileDevices
+from utilities import execute_command, secure_run_subprocess
+
+
+class LxdInstanceFiles:
+    """File manager of a LXD instance.
+
+    Attrs:
+        instance (LxdInstance): LXD instance where the files are located in.
+    """
+
+    def __init__(self, instance: LxdInstance):
+        """Construct the file manager.
+
+        Args:
+            instance: LXD instance where the files are located in.
+        """
+        self.instance = instance
+
+    def mk_dir(self, dir_name: str) -> None:
+        """Create a directory in the LXD instance.
+
+        Args:
+            dir: Name of the directory to create.
+        """
+        self.instance.execute(["/usr/bin/mkdir", "-p", dir_name])
+
+    def push(self, source: str, destination: str, mode: Optional[str] = None) -> None:
+        """Push a file to the LXD instance.
+
+        Args:
+            source: Path of the file to push to the LXD instance.
+            destination: Path in the LXD instance to load the file.
+            mode: File permission setting.
+
+        Raises:
+            LxdException: Unable to load the file into the LXD instance.
+        """
+        lxc_cmd = [
+            "/snap/bin/lxc",
+            "file",
+            "push",
+            source,
+            f"{self.instance.name}/{destination.lstrip('/')}",
+        ]
+
+        if mode:
+            lxc_cmd += ["--mode", str(mode)]
```

Comment:
```suggestion
            lxc_cmd += ["--mode", mode]
```
I think str conversion can be omitted :)

Reply:
There is already a comment addressing this, resolving, sorry about the duplicated issue!